src/components/interface/cbufp/cbuf_c.c

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2010 by The George Washington University.  All rights reserved.
+ *
+ * Redistribution of this file is permitted under the GNU General
+ * Public License v2.
+ *
+ * Author: Gabriel Parmer, gparmer@gwu.edu, 2010
+ * Updated by Qi Wang and Jiguo Song, 2011
+ * Updated and simplified by removing sub-page allocations, Gabe Parmer, 2012
+ */
+
+#include <cos_component.h>
+#include <print.h>
+#include <cbuf.h>
+#include <cbufp.h>
+#include <cbuf_vect.h>
+#include <cos_vect.h>
+#include <cos_debug.h>
+#include <cos_alloc.h>
+#include <valloc.h>
+
+#define CSLAB_ALLOC(sz)   alloc_page()
+#define CSLAB_FREE(x, sz) free_page(x)
+#include <cslab.h>
+
+/* 
+ * All of the structures that must be compiled (only once) into each
+ * component that uses cbufs.
+ *
+ * I'd like to avoid having to make components initialize this data,
+ * so try and use structures that by default are initialized as all
+ * zero. 
+ */
+extern struct cos_component_information cos_comp_info;
+
+CVECT_CREATE_STATIC(meta_cbuf);
+PERCPU_VAR(cbuf_alloc_freelists);
+
+/*** Slow paths for each cbuf operation ***/
+
+/* 
+ * Precondition: cbuf lock is taken.
```

Comment:
Update the comments please.  No more lock, so I don't think that a precondition is that the lock is taken.